### PR TITLE
[RNMobile] Refactor BlockToolbar follow-up PR

### PR DESCRIPTION
**Pull request by Tug**
## Description
This PR fixes some minor issues with https://github.com/WordPress/gutenberg/pull/16677

## How has this been tested?
Tested with https://github.com/wordpress-mobile/gutenberg-mobile/pull/1247

## Types of changes
UX changes:
- Hide keyboard when switching mode
- Do not force hide keyboard on android, as unselecting a block already handles it

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/editor/src/components/provider/index.native.js [hunk 1]
```diff
@@ -158,6 +160,7 @@ export default compose( [
 	} ),
 	withDispatch( ( dispatch ) => {
 		const {
+			clearSelectedBlock,
```
Comment (mchowning) [inline packages/editor/src/components/provider/index.native.js]: It looks like `clearSelectedBlock` from `core/editor` just [calls through](https://github.com/WordPress/gutenberg/blob/master/packages%2Feditor%2Fsrc%2Fstore%2Factions.js#L1021) to `core/block-editor` with a deprecation warning that shows up as a Yellow Box warning. Seems like we should just get this directly from `core/block-editor` instead.

Comment (Tug) [reply packages/editor/src/components/provider/index.native.js]: Thanks, updated 👍

Comment (mchowning) [reply packages/editor/src/components/provider/index.native.js]: I think we also need to update `header-toolbar`: https://github.com/WordPress/gutenberg/pull/16906/files#diff-dbd7e14b9675d5446422c008453093bbR91

**Review by mchowning — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**